### PR TITLE
removing kafka metrics processor service

### PR DESCRIPTION
**Pull request by gsps1**
JIRA https://issues.cask.co/browse/CDAP-12562

### CODE REVIEW

**Comment by chtyim** [comment]
LGTM

**Comment by ecfm** [comment]
There are some constants can be removed too, like `KAFKA_TOPIC_PREFIX`

File: cdap-watchdog/src/test/java/co/cask/cdap/metrics/process/MetricsProcessorServiceTest.java [hunk 1]
```diff
@@ -88,41 +74,14 @@ public void testMetricsProcessor() throws Exception {
     zkServer = InMemoryZKServer.builder().build();
```
Comment (ecfm) [inline cdap-watchdog/src/test/java/co/cask/cdap/metrics/process/MetricsProcessorServiceTest.java]: this can also be removed

**Comment by gsps1** [comment]
updated it @maochf, thanks

File: cdap-common/src/main/java/co/cask/cdap/common/conf/Constants.java [hunk 2]
```diff
@@ -597,9 +597,6 @@
     public static final String RETENTION_SECONDS = "metrics.data.table.retention.resolution";
 
     public static final String TOPIC_PREFIX = "metrics.topic.prefix";
-    public static final String KAFKA_TOPIC_PREFIX = "metrics.kafka.topic.prefix";
-    public static final String KAFKA_PARTITION_SIZE = "metrics.kafka.partition.size";
-    public static final String KAFKA_CONSUMER_PERSIST_THRESHOLD = "metrics.kafka.consumer.persist.threshold";
     public static final String KAFKA_META_TABLE = "metrics.kafka.meta.table";
```
Comment (ecfm) [inline cdap-common/src/main/java/co/cask/cdap/common/conf/Constants.java]: better to add a comment to clarify that this is a legacy name and Kafka is actually not used any more. The table stores meta data for TMS metrics topics

**Comment by ecfm** [comment]
`MetricsMetaKey` can also be removed since `TopicIdMetaKey` will be the only metric meta key

**Comment by gsps1** [comment]
@maochf i think, its better to keep the interface. no benefit from updating the methods from interface to class type. have added comment about the table name.

**Comment by ecfm** [comment]
LGTM

**Comment by gsps1** [comment]
build green: https://builds.cask.co/browse/CDAP-DUT6138-1 merging

**Pull request merged**